heudiconv/convert.py

```diff
@@ -1077,6 +1083,10 @@ def bvals_are_zero(bval_file: str) -> bool:
     True if all are all 0 or 5; False otherwise.
     """
 
+    # GE hyperband multi-echo containing diffusion info
+    if isinstance(bval_file, TraitListObject):
+        return all([bvals_are_zero(bvf) for bvf in bval_file])
```

Comment:
```suggestion
        return all(bvals_are_zero(bvf) for bvf in bval_file)
```
so we deal with generator and quit as soon as can . another alternative

```suggestion
        return all(map(bvals_are_zero, bval_file))
```